EBA/Taschenrechner/Calc/MathCalc.cs

```diff
@@ -95,64 +97,99 @@ public double root(int exp, double x)
+        /// Calculates the factorial from the Input
         /// </summary>
-        public int Fakultät(int X)
+        public int factorial(int X)
         {
             if (X == 0)
             {
                 return 1;
             }
-            return X * Fakultät(X-1);
+            return X * factorial(X - 1);
         }
+
         /// <summary>
         ///  Converts a float point number into a fraction
         ///  You only need to input num
+        ///  
+        ///  See here https://en.wikipedia.org/wiki/Euclidean_algorithm
         /// </summary>
         /// <returns> 2 Integers </returns>
-        public (int, int) DoubleToFraction(double num, double epsilon = 0.0001, int maxIterations = 50)
+        public (long, long) DoubleToFraction(double decimalSourceNumber, double precision = 0.0001, int maxIterations = 50)
         {
-            double[] d = new double[maxIterations + 2];
-            d[1] = 1;
-            double z = num;
-            double n = 1;
-            int t = 1;
+            double[] Denominators = new double[maxIterations + 2];
+            Denominators[0] = 0;
+            Denominators[1] = 1;
+            double decimalNumber = decimalSourceNumber;
+            double Numerator = 1;
+            long iterationCount = 1;
 
-            int wholeNumberPart = (int)num;
-            double decimalNumberPart = num - Convert.ToDouble(wholeNumberPart);
+            // that are the numbers befor the decimal point
+            long wholeNumberPart = (long)decimalSourceNumber;
+            // that are the numbers after the decimal point
+            double origninalDecimalNumberPart = decimalSourceNumber - Convert.ToDouble(wholeNumberPart);
 
-            while (t < maxIterations && ABS(n / d[t] - num) > epsilon)
+            // with each iteration this value comes closer to 0 but is never 0.
+            // rule: approximateValue > 0 
+            var approximateValue = ABS((Numerator / Denominators[iterationCount] + (long)decimalSourceNumber) - decimalSourceNumber);
+            while (iterationCount < maxIterations && approximateValue > precision)
             {
-                t++;
-                z = 1 / (z - (int)z);
-                d[t] = d[t - 1] * (int)z + d[t - 2];
-                n = (int)(decimalNumberPart * d[t] + 0.5);
+                iterationCount++;
+                wholeNumberPart = (long)decimalNumber;
+                var decimalNumberPart = decimalNumber - wholeNumberPart;
+                // the decimal number is now between 1 and lower 10
+                // because an excat match is 1 and a 0.9xxx is near 10 but never 10
+                decimalNumber = 1 / decimalNumberPart;
+                // a = q0 * b + r0
+                // b = q1 * r0 + r1
+                Denominators[iterationCount] = Denominators[iterationCount - 1] * (long)decimalNumber + Denominators[iterationCount - 2];
+
+                var WEDONTKNOW = origninalDecimalNumberPart * Denominators[iterationCount];
```

Comment:
here you can use X or something. That is a little bit better then wedontknow and tell the reader the same. :)